hazelcast/src/main/java/com/hazelcast/sql/SqlRow.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+/**
+ * SQL row.
+ */
+public interface SqlRow {
+    /**
+     * Gets the value of the column by index.
+     * <p>
+     * The class of the returned value depends on the SQL type of the column.
+     *
+     * @see #getRowMetadata()
+     * @see SqlColumnMetadata#getType()
+     * @param columnIndex Column index, 0-based.
+     * @return Value of the column.
+     * @throws IndexOutOfBoundsException If column index is out of bounds.
+     */
+    Object getObject(int columnIndex);
```

Comment:
would it make sense to provide getter for primitive types? or a method that supports generics so you don't need explicit casting? i.e  `<T> T get(int index)`

Reply:
Reverted back to `<T> T getObject(...)` approach, as it seems that the community is in agreement that it is more user-friendly. 
Also, added `<T> T getObject(String columnName)` overload for convenience.
There will be no specialized methods in the first release (e.g. `getInt(...)`) because it requires additional considerations on the desired behavior (e.g. how to deal with nullability, whether to perform implicit conversions, etc).